core/src/main/java/io/questdb/griffin/engine/join/HashJoinRecordCursorFactory.java

```diff
@@ -109,7 +108,8 @@ public boolean supportsUpdateRowId(CharSequence tableName) {
     }
 
     private void buildMapOfSlaveRecords(RecordCursor slaveCursor, SqlExecutionCircuitBreaker circuitBreaker) {
```

Comment:
It will be cleaner to move thid method inside the cursor, same as you did in `HashJoinLightRecordCursorFactory`